sn_auditor/src/dag_db.rs

```diff
@@ -228,39 +229,70 @@ impl SpendDagDb {
                 continue;
             }
 
-            // get a copy of the current DAG
-            let mut dag = { self.dag.clone().read().await.clone() };
-
-            // update it
-            client
-                .spend_dag_continue_from(&mut dag, addrs_to_get, spend_processing.clone(), true)
-                .await;
-
-            // update utxos
-            let new_utxos = dag.get_utxos();
-            utxo_addresses.extend(
-                new_utxos
-                    .into_iter()
-                    .map(|a| (a, Instant::now() + REATTEMPT_INTERVAL)),
-            );
+            #[cfg(not(feature = "dag-collection"))]
+            {
+                if let Some(sender) = spend_processing.clone() {
+                    // crawl DAG
+                    let tasks: Vec<_> = addrs_to_get
+                        .iter()
+                        .map(|a| client.spend_dag_crawl_from(*a, sender.clone()))
+                        .collect();
+                    let res = futures::future::join_all(tasks).await;
```

Comment:
Here the sender is cloned for each parallel crawl tasks in `sender.clone()`, meaning the single receiver will receive spends from all these crawling tasks concurrently.